Tests/src/PublicAPITests/SequenceSystemTests.cpp

```diff
@@ -0,0 +1,663 @@
+
+	// Create space
+	char UniqueSpaceName[256];
+	SPRINTF(UniqueSpaceName, "%s-%s", TestSpaceName, GetUniqueString().c_str());
+	const char* TestSpaceDescription = "CSP-UNITTEST-SPACEDESC-MAG";
+
+	csp::systems::Space Space;
+	CreateSpace(SpaceSystem, UniqueSpaceName, TestSpaceDescription, csp::systems::SpaceAttributes::Private, nullptr, nullptr, nullptr, Space);
+
+	// Enter space
+	auto [EnterResult] = AWAIT_PRE(SpaceSystem, EnterSpace, RequestPredicate, Space.Id);
+
+	bool CallbackCalled = false;
+
+	// Create sequence
+	csp::common::Array<csp::common::String> SequenceItems {"Hotspot1", "Hotspot2", "Hotspot3"};
+
+	const char* TestSequenceKey = "CSP-UNITTEST-SEQUENCE-MAG";
+	char UniqueSequenceName[256];
+	SPRINTF(UniqueSequenceName, "%s-%s", TestSequenceKey, GetUniqueString().c_str());
+
+	auto CreateCallback = [&CallbackCalled, &UniqueSequenceName](const csp::multiplayer::SequenceChangedParams& Params)
+	{
+		EXPECT_EQ(Params.Key, UniqueSequenceName);
+		EXPECT_EQ(Params.UpdateType, csp::multiplayer::ESequenceUpdateType::Create);
+
+		CallbackCalled = true;
+	};
+
+	Connection->SetSequenceChangedCallback(CreateCallback);
+
+	csp::systems::Sequence Sequence;
+	CreateSequence(SequenceSystem, UniqueSequenceName, "GroupId", Space.Id, SequenceItems, Sequence);
+
+	WaitForCallback(CallbackCalled);
+	EXPECT_TRUE(CallbackCalled);
+
+	// Rename sequence
+	const char* TestUpdatedSequenceKey = "CSP-UNITTEST-SEQUENCE-MAG-UPDATED";
+	char UniqueUpdatedSequenceName[256];
+	SPRINTF(UniqueUpdatedSequenceName, "%s-%s", TestUpdatedSequenceKey, GetUniqueString().c_str());
+
+	auto UpdateCallback = [&CallbackCalled, &Sequence, &UniqueUpdatedSequenceName](const csp::multiplayer::SequenceChangedParams& Params)
+	{
+		EXPECT_EQ(Params.Key, Sequence.Key);
+		EXPECT_EQ(Params.UpdateType, csp::multiplayer::ESequenceUpdateType::Update);
+		EXPECT_EQ(Params.NewKey, std::string(UniqueUpdatedSequenceName).c_str());
+
+		CallbackCalled = true;
+	};
+
+	Connection->SetSequenceChangedCallback(UpdateCallback);
+	CallbackCalled = false;
+
+	csp::systems::Sequence UpdatedSequence;
+	RenameSequence(SequenceSystem, UniqueSequenceName, UniqueUpdatedSequenceName, UpdatedSequence);
+
+	WaitForCallback(CallbackCalled);
+	EXPECT_TRUE(CallbackCalled);
+
+	// Delete sequence
+	auto DeleteCallback = [&CallbackCalled, &UniqueUpdatedSequenceName](const csp::multiplayer::SequenceChangedParams& Params)
+	{
+		EXPECT_EQ(Params.Key, UniqueUpdatedSequenceName);
+		EXPECT_EQ(Params.UpdateType, csp::multiplayer::ESequenceUpdateType::Delete);
+
+		CallbackCalled = true;
+	};
+
+	Connection->SetSequenceChangedCallback(DeleteCallback);
+	CallbackCalled = false;
+
+	DeleteSequences(SequenceSystem, {UpdatedSequence.Key});
+
+	WaitForCallback(CallbackCalled);
+	EXPECT_TRUE(CallbackCalled);
+
+	SpaceSystem->ExitSpace(
+		[](const csp::systems::NullResult& Result)
+		{
+		});
+
+	// Delete space
+	DeleteSpace(SpaceSystem, Space.Id);
+
+	// Log out
+	LogOut(UserSystem);
+}
+
+#endif
```

Comment:
Is it possible to test the case where DeleteSequences will fail because  `This call will fail if the user isn't a creator of the space`

Reply:
Added a test which tests sequence key change, edit, and delete from a user not in the space